### PR TITLE
Enhance GoDaddyProvider.ListZonesAsync method to use marker argument …

### PR DESCRIPTION
**Pull request by johnburns320**
…to get paged list of zones #581

### CODE REVIEW

File: KeyVault.Acmebot/Providers/GoDaddyProvider.cs [hunk 1]
```diff
@@ -81,13 +81,31 @@ public GoDaddyClient(string apiKey, string apiSecret)
...
 
-            var domains = await response.Content.ReadAsAsync<ZoneDomain[]>();
+                if (domains.Length == limit)
+                {
+                    if (domains.Last().Domain is not null)
+                    {
+                        marker = $"&marker={domains.Last().Domain}";
+                    }
+                }
+            } while (marker != string.Empty);
```
Comment (shibayan) [inline KeyVault.Acmebot/Providers/GoDaddyProvider.cs]: I'm not familiar with the GoDaddy API, but could this be accomplished with code as simple as this?

```csharp
            var limit = 100;
            var marker = "";
            var allActiveDomains = new List<ZoneDomain>();

            while (true)
            {
                var response = await _httpClient.GetAsync($"v1/domains?statuses=ACTIVE&includes=nameServers&limit={limit}{marker}");

                response.EnsureSuccessStatusCode();

                var domains = await response.Content.ReadAsAsync<ZoneDomain[]>();

                if (domains.Length == 0)
                {
                    break;
                }

                allActiveDomains.AddRange(domains);

                marker = $"&marker={domains[^1].Domain}";
            }
```

**Review by shibayan — CHANGES_REQUESTED** [review]

File: KeyVault.Acmebot/Providers/GoDaddyProvider.cs (hunk 1 above)
Comment (johnburns320) [reply KeyVault.Acmebot/Providers/GoDaddyProvider.cs]: Looks good to me!

**Review by shibayan — APPROVED** [review]

**Comment by shibayan** [comment]
Thanks!

**Comment by apmantc** [comment]
Is it possible to get this fixed rolled into the documented latest update file?:
https://stacmebotprod.blob.core.windows.net/keyvault-acmebot/v4/latest.zip

**Comment by shibayan** [comment]
Not yet released. It will be released soon.

**Pull request merged**